### PR TITLE
Add more default plot settings to workbench

### LINKED ISSUE #29739: More default plot settings (ticks, axes, grid)
**Original reporter:** Max at ISIS

It would be useful to be able to set more default options for 2D plots:
- customise ticks (I'm not sure what Max means here - maybe size and line width, but we should check with him)
- tick labels (whether they are in/out, and whether to display on top/bottom/left/right axes)
- axis line width
- whether grid is enabled by default

### PR DESCRIPTION
**Pull request by joseph-torsney**
This PR adds more default plot settings to workbench:
- Axes spine width
- Whether grid is turned on
- Enable/disable ticks on each side of the plot
- Enable/disable tick labels on each side of the plot
- Set width, length, and direction for major and minor ticks

**To test:**
- Ensure enabling/disabling all of the settings above work correctly.
- Are the changes to the gui clear and easy to use in your opinion? 
- Should we add some of these settings to the plot config? For example, a tiled plot with ticks and labels on all sides can look a bit messy!
image

Fixes #29739 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/concepts/PropertiesFile.rst [hunk 1]
```diff
@@ -276,6 +272,42 @@ Plotting Settings
...
+|``plots.showLabelsRight``        |Whether to show labels on the right side of the plot              | ``On``, ``Off``     |
++---------------------------------+------------------------------------------------------------------+---------------------+
+|``plots.showLabelsTop``          |Whether to show labels on the top side of the plot                | ``On``, ``Off``     |
++---------------------------------+------------------------------------------------------------------+---------------------+
+|``plots.ticks.major.length``     |The default length of the major ticks                             |``6``                |
++---------------------------------+------------------------------------------------------------------+---------------------+
+|``plots.ticks.major.width``      |The default width of the major ticks                              |``1``                |
++---------------------------------+------------------------------------------------------------------+---------------------+
+|``plots.ticks.major.direction``  |The default direction of the major ticks                          |``In``, ``Out``      |
```
Comment (Harrietbrown) [inline docs/source/concepts/PropertiesFile.rst:303]: both this and the minor ticks can also be ``InOut``

**Review by Harrietbrown — CHANGES_REQUESTED** [review]
I think this wiould benefit from a button to restore the dafault plot settings to factory default.

**Comment by martyngigg** [comment]
@Harrietbrown We're keen to getting the settings changes in and have created a new issue for resetting to defaults.

@joseph-torsney Could you address the point that Harriet made in the docs about the ticks can also have the `InOut` option and then we are good to go.

**Comment by martyngigg** [comment]
Unrelated test failures and minor docs correction after first review.

**Pull request merged**